File: python/OECD-Common/GenderEnt1.py
class GenderEnt1:
    Id: int

    CountryId: str
    CountryName: str
    IndicatorId: str
    IndicatorName: str
    SexId: str
    SexName: str
    AgeId: str
    AgeName: str

    TimeFormatId: str
    TimeFormatName: str
    UnitId: str
    UnitName: str
    UnitMultiplierId: str
    UnitMultiplierName: str
    ReferencePeriodId: str
    ReferencePeriodName: str

    Year: int
    Value: float
    Status: str

    # https://stackoverflow.com/questions/35282222/in-python-how-do-i-cast-a-class-object-to-a-dict
    def asdict(self):
        obj_dict = {}
        if hasattr(self, "Id"):
            obj_dict["Id"] = self.Id
        obj_dict.update({
            "CountryId": self.CountryId,
            "CountryName": self.CountryName,
            "IndicatorId": self.IndicatorId,
            "IndicatorName": self.IndicatorName,
            "SexId": self.SexId,
            "SexName": self.SexName,
            "AgeId": self.AgeId,
            "AgeName": self.AgeName,

            "TimeFormatId": self.TimeFormatId,
            "TimeFormatName": self.TimeFormatName,
            "UnitId": self.UnitId,
            "UnitName": self.UnitName,
            "UnitMultiplierId": self.UnitMultiplierId,
            "UnitMultiplierName": self.UnitMultiplierName,
            "ReferencePeriodId": self.ReferencePeriodId if hasattr(self, "ReferencePeriodId") else None,
            "ReferencePeriodName": self.ReferencePeriodName if hasattr(self, "ReferencePeriodName") else None,

            "Year": self.Year,
            "Value": self.Value,
            "Status": self.Status if hasattr(self, "Status") else None
        })
        return obj_dict


class GenderEnt1Dimension:
    CountryId: str
    CountryName: str
    IndicatorId: str
    IndicatorName: str
    SexId: str
    SexName: str
    AgeId: str
    AgeName: str


class GenderEnt1Attribute:
    TimeFormatId: str
    TimeFormatName: str
    UnitId: str
    UnitName: str
    UnitMultiplierId: str
    UnitMultiplierName: str
    ReferencePeriodId: str
    ReferencePeriodName: str


class GenderEnt1Observation:
    Year: int
    Value: float
    Status: str

    def __init__(self, year, value, status):
        self.Year = year
        self.Value = value
        self.Status = status

File: csharp/OECD-ETL/Oecd.Data.Common/GenderEnt1.cs
namespace Oecd.Common;

public class GenderEnt1 : IGenderEnt1Dimension, IGenderEnt1Attribute, IGenderEnt1Observation
{
    public GenderEnt1() { }

    public GenderEnt1(IGenderEnt1Dimension dimension, IGenderEnt1Attribute attribute, IGenderEnt1Observation observation)
    {
        SetDimensionProperties(dimension);
        SetAttributeProperties(attribute);
        SetObservationProperties(observation);
    }

    public int Id { get; set; }

    #region IGenderEnt1Dimension
    public string CountryId { get; set; } = null!;
    public string CountryName { get; set; } = null!;
    public string IndicatorId { get; set; } = null!;
    public string IndicatorName { get; set; } = null!;
    public string SexId { get; set; } = null!;
    public string SexName { get; set; } = null!;
    public string AgeId { get; set; } = null!;
    public string AgeName { get; set; } = null!;
    #endregion

    #region IGenderEnt1Attribute
    public string TimeFormatId { get; set; } = null!;
    public string TimeFormatName { get; set; } = null!;
    public string UnitId { get; set; } = null!;
    public string UnitName { get; set; } = null!;
    public string UnitMultiplierId { get; set; } = null!;
    public string UnitMultiplierName { get; set; } = null!;
    public string? ReferencePeriodId { get; set; }
    public string? ReferencePeriodName { get; set; }
    #endregion

    #region IGenderEnt1Observation
    public int Year { get; set; }
    public double? Value { get; set; }
    public string? Status { get; set; }
    #endregion

    public void SetDimensionProperties(IGenderEnt1Dimension dimension)
    {
        CountryId = dimension.CountryId;
        CountryName = dimension.CountryName;
        IndicatorId = dimension.IndicatorId;
        IndicatorName = dimension.IndicatorName;
        SexId = dimension.SexId;
        SexName = dimension.SexName;
        AgeId = dimension.AgeId;
        AgeName = dimension.AgeName;
    }

    public void SetAttributeProperties(IGenderEnt1Attribute attribute)
    {
        TimeFormatId = attribute.TimeFormatId;
        TimeFormatName = attribute.TimeFormatName;
        UnitId = attribute.UnitId;
        UnitName = attribute.UnitName;
        UnitMultiplierId = attribute.UnitMultiplierId;
        UnitMultiplierName = attribute.UnitMultiplierName;
        ReferencePeriodId = attribute.ReferencePeriodId;
        ReferencePeriodName = attribute.ReferencePeriodName;
    }

    public void SetObservationProperties(IGenderEnt1Observation observation)
    {
        Year = observation.Year;
        Value = observation.Value;
        Status = observation.Status;
    }
}

public interface IGenderEnt1Dimension
{
    public string CountryId { get; set; }
    public string CountryName { get; set; }
    public string IndicatorId { get; set; }
    public string IndicatorName { get; set; }
    public string SexId { get; set; }
    public string SexName { get; set; }
    public string AgeId { get; set; }
    public string AgeName { get; set; }
}

public interface IGenderEnt1Attribute
{
    public string TimeFormatId { get; set; }
    public string TimeFormatName { get; set; }
    public string UnitId { get; set; }
    public string UnitName { get; set; }
    public string UnitMultiplierId { get; set; }
    public string UnitMultiplierName { get; set; }
    public string? ReferencePeriodId { get; set; }
    public string? ReferencePeriodName { get; set; }
}

public interface IGenderEnt1Observation
{
    public int Year { get; set; }
    public double? Value { get; set; }
    public string? Status { get; set; }
}

public class GenderEnt1Dimension : IGenderEnt1Dimension
{
    public string CountryId { get; set; } = null!;
    public string CountryName { get; set; } = null!;
    public string IndicatorId { get; set; } = null!;
    public string IndicatorName { get; set; } = null!;
    public string SexId { get; set; } = null!;
    public string SexName { get; set; } = null!;
    public string AgeId { get; set; } = null!;
    public string AgeName { get; set; } = null!;
}

public class GenderEnt1Attribute : IGenderEnt1Attribute
{
    public string TimeFormatId { get; set; } = null!;
    public string TimeFormatName { get; set; } = null!;
    public string UnitId { get; set; } = null!;
    public string UnitName { get; set; } = null!;
    public string UnitMultiplierId { get; set; } = null!;
    public string UnitMultiplierName { get; set; } = null!;
    public string? ReferencePeriodId { get; set; }
    public string? ReferencePeriodName { get; set; }
}

public class GenderEnt1Observation : IGenderEnt1Observation
{
    public int Year { get; set; }
    public double? Value { get; set; }
    public string? Status { get; set; }
}
